.github/workflows/irso-functional.yml

```diff
@@ -0,0 +1,49 @@
+name: IrSO Functional Tests
+
+on:
+  pull_request:
```

Comment:
In [BMO we have](https://github.com/metal3-io/baremetal-operator/blob/24cecbedb46e09bccd55629977098ba38797f4fb/.github/workflows/pipeline.yml#L5) also
` types: [opened, edited, synchronize, reopened, ready_for_review]`

By [default](https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#pull_request), `pull_request` triggers only on `opened, synchronize, or reopened`.

I think we should be consistent in our repos for better develover experience, so either this should have the expanded type triggers or we should simplify BMO/CAPM3/IPAM.  

I see IRSO improvement PR also merged just this simple trigger, so same should be applied there.

This can also be done in follow-up, no need to block this.

Reply:
"edited" seems strange: it will trigger on e.g. editing the PR title?

"ready_for_review" probably makes sense

Reply:
I cross-checked also CAPM3, there we have yet another set of types, so indeed we should make it consistent there too. 

`edited` is necessary for some of the checks like PR verifier and DCO which inteact with the PR metadata, but having `edited` with `pull_request` is just re-running stuff for no reason. It was probably added as a workaround in BMO to the Prow issue of lacking approval after repushing to branch for external contributors, and we did not have the PR auto-reapprover workflow yet at the time.

 I suppose we agree the types list would `opened, synchronize, reopened, ready_for_review'  ?

Reply:
Looks reasonable to me

Reply:
I'll push some PRs to make it consistent across the repos, if you update this one.

Reply:
Made following to make triggers match in all of the other repos.
- https://github.com/metal3-io/cluster-api-provider-metal3/pull/2318
- https://github.com/metal3-io/ip-address-manager/pull/885
- https://github.com/metal3-io/baremetal-operator/pull/2242
- https://github.com/metal3-io/ironic-standalone-operator/pull/127